src/core/mount.c

```diff
@@ -906,6 +908,7 @@ static void mount_enter_unmounting(Mount *m) {
 
 fail:
         log_unit_warning_errno(UNIT(m), r, "Failed to run 'umount' task: %m");
+        m->stop_result = MOUNT_FAILURE_RESOURCES;
         mount_enter_mounted(m, MOUNT_FAILURE_RESOURCES);
```

Comment:
so, both "stop_result" and "result" will no be affected by this failure. I am not sure this is right? Or is it? I am not seeing clearly here... so far we only had one result code, hence things where easy, but if we have two, shouldn't this be decoupled fully? Can you explain your thoughts on this?

Reply:
I was thinking `result` still needed to be affected.  Previously, the overall service state was set to "failed", and this would be visible in `systemctl status.  I wanted to leave that behaviour unchanged.